pkg/buffer/buffer.go

```diff
@@ -132,13 +133,13 @@ func (buffer *Buffer) Reset() {
 	buffer.events = []*add_events.Event{}
 
 	buffer.bufferSent = time.Now()
-	buffer.lenThreads = 0
-	buffer.lenLogs = 0
-	buffer.lenEvents = 0
+	buffer.lenThreads = atomic.Int32{}
```

Comment:
Yeah, looks like that int32 is the smallest supported type by the official sync package.

It looks like there is non-upstream implementation (https://github.com/hslam/atomic) which also supports smaller types which may save us some bytes, but this is really over optimization and not needed (just me thinking out loud and getting familiar with those types and APIs).

Reply:
It's little bit more complicated - https://go101.org/article/memory-layout.html - there is also some padding involved.